ensime-editor.el

```diff
@@ -583,6 +583,19 @@ Returns a function/closure to invoke the necessary buffer operations to perform
            insert save-excursion)
       (ensime-indent-line))))
 
+(defun ensime-move-to-end-of-multiline-import ()
+  "Move point to the end of a multiline import."
+  (when (not (string-match (rx (zero-or-more any) ?} (zero-or-more whitespace) eol) (thing-at-point 'line t)))
+      (goto-char (point-at-eol))
+      (when (not (equal (point) (point-max)))
+        (forward-line 1)
+        (ensime-move-to-end-of-multiline-import)
+      )))
+
+(defun ensime-at-start-of-multiline-block ()
+  "Returns t when at start of multiline block."
+      (string-match (rx (zero-or-more anything) ?{ (zero-or-more (not (any ?})))) (thing-at-point 'line t)))
```

Comment:
I personally find regex easier to read but I'm ok with this. Have you seen the pcre2el package?

I believe the more standard thing to do is to piggy back off the face highlighting of the buffer but perhaps correctness is more important here than performance.